src/middleware/packages/triplestore/index.js

```diff
@@ -129,7 +129,7 @@ const TripleStoreService = {
         body: 'update=DROP+ALL',
         headers: {
           'Content-Type': 'application/x-www-form-urlencoded',
-          'X-SemappsUser': 'URIofUser',
+          'X-SemappsUser': params.webId,
```

Comment:
Comme le webid est dans le `ctx` (ctx.meta.webId) et qu'il est ajouté automatiquement par les fonctions authorize ou authenticate, je me demande si on ne pourrait pas laisser le service triplestore le récupérer directement, plutôt que de passer ce paramètre en cascade ?
Pour moi l'objet `ctx` a justement ce rôle de variable "globale" qui permet de définir le *contexte* de l'appel. En tout cas dans React, le Context a pour but de "faire passer des données à travers l’arborescence du composant sans avoir à passer manuellement les props à chaque niveau" (https://fr.reactjs.org/docs/context.html)

Reply:
tres bonne idee. le context nous suit au travers des appels imbriques.
attention abien faire des ctx.call() et pas des this.broker.call() or broker.call() si on veut que le context nous suive. seul `ctx.call` fait suivre le context

Reply:
Ok sur le principe mais ça pose un problème pour les appels des action LDP. Si un service tiers veut appeler une action ldp, il passe un paramètre webId "classique". Dans la proposition que vous faite, cela voudrait dire que l'appelant doit passer un paramètre dans le meta. En plus du fait que cela ne ma semble pas développer friendly, ce n'est pas contrôlable par les validation de paramètres.

Reply:
Bien vu, je suis OK pour permettre de passer le webID comme paramètre optionel de l'action, mais que sinon le service se base sur le contexte.

Reply:
geré de manière globale comme les accept et Content-Type